libmscore/utils.cpp

```diff
@@ -760,9 +760,9 @@ Note* searchTieNote(Note* note)
                   QList<Chord*> gnb = c->graceNotesBefore();
                   if (!gnb.isEmpty()) {
                         Chord* gc = gnb[0];
-                        note2 = gc->findNote(note->pitch());
```

Comment:
I noticed that too, but it happens before the loop, so it shouldn't interfere with anything.  Ditto with line 722.  The problem was with a good value of note2 (calculated on a previous iteration) being clobbered by line 763.